cpp/src/utilities/cython.cpp

```diff
@@ -27,28 +28,159 @@ namespace cython {
 // Populates a graph_container_t with a pointer to a new graph object and sets
 // the meta-data accordingly.  The graph container owns the pointer and it is
 // assumed it will delete it on destruction.
-//
-// FIXME: Should local_* values be void* as well?
 void populate_graph_container(graph_container_t& graph_container,
-                              legacyGraphTypeEnum legacyType,
-                              raft::handle_t const& handle,
-                              void* offsets,
-                              void* indices,
+                              raft::handle_t& handle,
+                              void* src_vertices,
+                              void* dst_vertices,
                               void* weights,
-                              numberTypeEnum offsetType,
-                              numberTypeEnum indexType,
+                              void* vertex_partition_offsets,
+                              numberTypeEnum vertexType,
+                              numberTypeEnum edgeType,
                               numberTypeEnum weightType,
-                              int num_vertices,
-                              int num_edges,
-                              int* local_vertices,
-                              int* local_edges,
-                              int* local_offsets,
+                              int num_partition_edges,
+                              int num_global_vertices,
+                              int num_global_edges,
+                              int partition_row_size,  // pcols
+                              int partition_col_size,  // prows
                               bool transposed,
                               bool multi_gpu)
 {
   CUGRAPH_EXPECTS(graph_container.graph_ptr_type == graphTypeEnum::null,
                   "populate_graph_container() can only be called on an empty container.");
 
+  bool do_expensive_check{false};
+  bool hypergraph_partitioned{false};
+
+  raft::comms::comms_t const& communicator = handle.get_comms();
+  int const rank = communicator.get_rank();
+  int partition_row_rank = rank / partition_row_size;
+  int partition_col_rank  = rank % partition_row_size;
+
+  // Setup the subcommunicators needed for this partition on the handle
+  partition_2d::subcomm_factory_t<partition_2d::key_naming_t, int>
+    subcomm_factory(handle, partition_row_size);
```

Comment:
And shouldn't we better initialize subcomms right after initializing comms? Why are we delaying subcommunicator initialization to this point?

Reply:
I've added a FIXME for this in the next commit.